bsc.verilog/undet/undet.exp

```diff
@@ -11,3 +11,19 @@ if { $vtest == 1 } {
     find_regexp sysUndetIf.v {assign d\$D\_IN \= p2 \? d \: i \;}
 }
 
+# Tests if-else-tree of constants simplification.
+if { $vtest == 1 } {
+    compile_verilog_pass UndetComp.bs
+    # Make sure 13-bit y values are optimized away.
+    string_does_not_occur sysUndetComp.v "13'd"
```

Comment:
If the Verilog referenced 13-bit values, but with a different base, this check would silently pass.

Reply:
True. The test is relying on the compiler preserving the base of the user's literals (which it's usually pretty good at).

Do you think it would be better to search for for "13'" or a regexp covering 13'd, 13'h and 13'b?

Reply:
Or should I be looking for the `[12 : 0]` size declaration for the 13-bit temporary that shouldn't exist?

Reply:
I'm OK with whatever you want, even the original check -- I just wanted to point it out.  I might have gone with `13'`. Checking the temporary seems good, but also the Verilog might not declare a temporary (it might just be inlined).

I wouldn't be opposed to just comparing the entire Verilog file against an expected file -- then, if _anything_ changes, the user is alerted and can check.  This has the chance of being helpful when someone is working on unrelated changes to the Verilog, too, although I recognize that it can also be a hassle to update lots of files when some global (possibly cosmetic) change is made (and that it can add extra commits in the history for this test).

Reply:
Given that the Verilog still has unnecessary != checks (presumably from some limitation in ITransform), I don't think it is clean enough to check directly. Other than that, I'm fine with the test as-is, but I'd also be fine if you changed it.

Reply:
What do you mean by "check directly"?

I think I would just replace all of the string checks with `compare_file_pass` of the Verilog that is currently generated (not against some ideal that we want), with a comment explaining that there should be no 13-bit `y` values (which there aren't) and should be no `!=` comparisons (which still needs to be fixed, when there are undefined values before the end).  (If someone makes a change that causes the comparison to fail, they'll have the context to know how to resolve it: either see that the situation improved, see that it regressed, or see that an unrelated changed happened and the expected file just needs to be updated.)